pkg/dispatcher/dispatcher/retry.go

```diff
@@ -58,13 +60,21 @@ func (d *Dispatcher) Dispatch(event *cloudevents.Event, subscription Subscriptio
 	var deadLetterURL *url.URL
 	if subscription.Delivery != nil && subscription.Delivery.DeadLetterSink != nil && !subscription.Delivery.DeadLetterSink.URI.IsEmpty() {
 		// TODO - Currently ignoring dead-letter configuration due to wrapping retry implementation - would send one deadletter for every retry :(
-		// deadLetterURL = subscription.Delivery.DeadLetterSink.URI.URL()
-		logger.Warn("Subscription Delivery DeadLetterSink Not Currently Supported!")
+		deadLetterURL = subscription.Delivery.DeadLetterSink.URI.URL()
+		// logger.Warn("Subscription Delivery DeadLetterSink Not Currently Supported!")
 	}
 
 	// Attempt To Dispatch The CloudEvent Message Via Knative Message Dispatcher With Retry Wrapper
 	err := retryRunner.Run(context.Background(), func(ctx context.Context) error {
```

Comment:
FYI, the cloudevent SDK supports retries: https://github.com/cloudevents/sdk-go/blob/master/v2/protocol/http/protocol_retry.go#L16

Do you think you could leverage it?